### PR TITLE
transformer plugin interface

### PR DESCRIPTION
**Pull request by kensou97**
This pr adds some constants and interfaces used to implement a gp-stream-server `transformer` plugin.
It also contains a simple example.

### CODE REVIEW

**Review by flykos — APPROVED** [review]
It looks like a so file will be compiled and deployed to the specified directory

**Comment by kensou97** [comment]
> It looks like a so file will be compiled and deployed to the specified directory

Yes, customer compiles and deploys the plugin so. Then specify the so path in job config file.

File: transformer/README.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+In this section, we'll use an example to show how to write your own transformer plugin. Our example transformer is [simple_transformer.go](./examples/simple_transformer.go). The implementation is really simple, it takes json data like `{"A":2,"B":1}` as input and takes an operator from properties.If operator=="+", it puts C=A+B back, if operator=='-', it puts C=A-B back.
+
+## 1 Implement `init` and `transform` function
+
+There are two main functions to be implemented. We call them `init` function and `transform` function,  this doesn't mean the function must have name "init" or "transform", you can name your function according to your needs.
+
+###  1.1 init function
+
+Init function will be invoked when the transformer plugin is first loaded. It's guaranteed to be invoked only once even if the plugin is used in multi jobs.
```
Comment (huiliang-liu) [inline transformer/README.md]: Add the description that Init function is not mandatory

Comment (kensou97) [reply transformer/README.md]: > Add the description that Init function is not mandatory

fixed

**Review by huiliang-liu — APPROVED** [review]

**Pull request merged**